### ISSUE TITLE
[BUG] warnings: Marlin/src/sd/cardreader.cpp: In static member function 'static void CardReader::presort()':

### ISSUE THREAD

**Issue by sl1pkn07** [MarlinFirmware/Marlin#14144]
### Description

warnings about build cardreader.cpp

### Steps to Reproduce

1. [First Step]
download marlin 2.0 from git (df7733425cda02f7e88e7eb9496be72a17d335c1)
2. [Second Step]
edit the default configuration scripts, you can apply this commit https://github.com/sl1pkn07/Marlin/commit/7441c1500e02ebffb35e3f3a080cf218633ba44e with the all changes 
3. [and so on...]
start build with Atom+PlatformIO (linux), all up to date

**Expected behavior:** [What you expect to happen]

build without warnings (warning about external components is alloweb because is external)

**Actual behavior:** [What actually happens]
~~~
Compiling .pioenvs/LPC1768/lib8c6/TMCStepper/source/SW_MODE.cpp.o
Marlin/src/sd/cardreader.cpp: In static member function 'static void CardReader::presort()':
Marlin/src/sd/cardreader.cpp:765:72: warning: array subscript 27 is above array bounds of 'char [27]' [-Warray-bounds]
                                     sortnames[I][SORTED_LONGNAME_MAXLEN] = '\0'; }while(0)
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
Marlin/src/sd/cardreader.cpp:942:11: note: in expansion of macro 'SET_SORTNAME'
           SET_SORTNAME(0);
           ^~~~~~~~~~~~
Marlin/src/sd/cardreader.cpp:765:72: warning: array subscript 27 is above array bounds of 'char [27]' [-Warray-bounds]
                                     sortnames[I][SORTED_LONGNAME_MAXLEN] = '\0'; }while(0)
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
Marlin/src/sd/cardreader.cpp:852:13: note: in expansion of macro 'SET_SORTNAME'
             SET_SORTNAME(i);
             ^~~~~~~~~~~~
~~~

#### Additional Information

* Include a ZIP file containing your `Configuration.h` and `Configuration_adv.h` files.
https://github.com/sl1pkn07/Marlin/commit/7441c1500e02ebffb35e3f3a080cf218633ba44e
* Provide pictures or links to videos that clearly demonstrate the issue.
![Screenshot_20190526_192531](https://user-images.githubusercontent.com/462213/58385143-0ab2a500-7fec-11e9-9617-3ba9545f5fb1.png)
* See [How Can I Contribute](#how-can-i-contribute) for additional guidelines.

**Comment by chrisqwertz** [comment]
Looks like an off by one error to my. I think that correct way would to replace `SORTED_LONGNAME_MAXLEN` with `SORTED_LONGNAME_MAXLEN - 1` in the places where the errors are thrown

**Comment by robbycandra** [comment]
Fixed with PR #14148

**Comment by sl1pkn07** [comment]
hi

yep, fixed by #14148 

greetigs and closing
